xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-ui/src/main/resources/CKEditor/FileUploader.xml

```diff
@@ -103,12 +103,21 @@
       #set ($document = $xwiki.getDocument($request.document))
     #end
     #set ($reference = $document.documentReference)
-    #set ($attachment = $services.temporaryAttachments.uploadTemporaryAttachment($reference, 'upload', 
+    #try('uploadTemporaryAttachmentException')
+      #set ($attachment = $services.temporaryAttachments.uploadTemporaryAttachment($reference, 'upload',
         $request.filename))
+    #end
     #if ($attachment)
       #sendSuccess($document, $attachment.filename)
+    #elseif ("$!uploadTemporaryAttachmentException" != '')
+      ## The exception is wrapped in an org.apache.velocity.exception.MethodInvocationException
+      #if ($uploadTemporaryAttachmentException.cause.translationKey)
+        #sendError(400 "$services.localization.render($uploadTemporaryAttachmentException.cause.translationKey)")
+      #else
+        #sendError(400 "$services.localization.render('ckeditor.upload.error.emptyReturn')")
+      #end
     #else
-      #sendError(400 "$services.localization.render('ckeditor.upload.error.emptyreturn')")
+      #sendError(400 "$services.localization.render('ckeditor.upload.error.emptyReturn')")
```

Comment:
Is this translation key change really expected ?

Reply:
This is an unrelated bug that I've fixed along the way. I should probably commit that separately.